lib/ddtrace/contrib/active_support/notifications/event.rb

```diff
@@ -56,6 +56,40 @@ def tracer
             end
           end
         end
+
+        # Extension to {Event} class that ensures the current {Context}
+        # is always clean when the event is processed.
+        #
+        # This is a safeguard as Contexts are thread-bound.
+        # If an integration re-uses threads, the context from a previous
+        # execution could leak into the new execution.
+        #
+        # This module *cannot* be used for events can be nested, as
+        # it drops all spans currently active in the {Context}.
+        module RootEvent
```

Comment:
Could we add unit tests for this module specifically? I'd like to see some assurance that if you use this module, it will prevent contexts from leaking.

I saw there's a test in Racecar, but I'd prefer something more explicit and directly related to this module, as its the component that owns this behavior.